Userland/Libraries/LibCore/LocalSocket.cpp

```diff
@@ -53,6 +53,17 @@ LocalSocket::~LocalSocket()
 {
 }
 
+ucred LocalSocket::peer_creds() const
+{
+    struct ucred creds = {};
+    socklen_t creds_size = sizeof(creds);
+    int rc = getsockopt(fd(), SOL_SOCKET, SO_PEERCRED, &creds, &creds_size);
```

Comment:
If you don't want to do error handling, maybe at least `dbgln`(or `perror`) what error occures (if any), so it's easier to find out what happened. More optimal would probably be to return `Result<ucred, int>` or `Result<ucred, OSError>` and allow the caller to do the error handling.

Reply:
`getsockopt` will only fail if the socket doesn't exist or if its other inputs are incorrect, so the only possibility will be an invalid fd. I think a `LocalSocket` with an invalid fd doesn't make sense so crashing is the best we can do.

I have added the error message :)

Reply:
Yes, I didn't consider that. I somehow thought about `pledge`, but if you don't pledge "unix", but have a `LocalSocket`, I don't know what you're up to, so that shouldn't be a problem too.